### PR TITLE
Revert: Revert "Set val escape on expression variables (#64414)"

### PR DESCRIPTION
**Pull request by cston**
Commit 1: Revert: Revert "Set val escape on expression variables (#64414)"
Commit 2: Update escape analysis to ignore local declarations only.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs:654]: We have other cases where we check `DeclarationKind: not BoundLocalDeclarationKind.None`. In future we might want to be consistent here, or maybe introduce an extension/computed property which asks if the bound-local represents a user-visible variable declaration, which throws when an unknown DeclarationKind is encountered.

No need to change anything in this PR.

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs:8353]: nit: consider indenting this to match the previous test baseline

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs
Comment (cston) [reply src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs:8353]: I'll update the indenting in a follow-up PR, unless other changes are needed in this PR.

**Comment by RikkiGibson** [comment]
The [second commit](https://github.com/dotnet/roslyn/pull/64633/commits/a4f21f09ec5e0642830e02d4be973310752e296d) needs a second review.

**Review by jaredpar — APPROVED** [review]

**Pull request merged**
